mcp/src/main/java/io/modelcontextprotocol/spec/McpServerSession.java

```diff
@@ -134,7 +134,8 @@ public <T> Mono<T> sendRequest(String method, Object requestParams, TypeReferenc
 	@Override
 	public Mono<Void> sendNotification(String method, Object params) {
 		McpSchema.JSONRPCNotification jsonrpcNotification = new McpSchema.JSONRPCNotification(McpSchema.JSONRPC_VERSION,
-				method, params);
+				method, this.transport.unmarshalFrom(params, new TypeReference<Map<String, Object>>() {
```

Comment:
I think we don't need
```java
this.transport.unmarshalFrom(params, new TypeReference<Map<String, Object>>() {})
```

and we can just provide `params` directly.

I'll commit this change.